figanos/matplotlib/plot.py

```diff
@@ -570,55 +576,56 @@ def gridmap(
         plot_kw["cbar_kwargs"].setdefault("label", wrap_text(cbar_label))
 
     # plot
+    plotting = {"transform": transform, "cmap": cmap, **plot_kw}
+    if ax:
+        plotting.setdefault("ax", ax)
     if contourf is False:
-        im = plot_data.plot.pcolormesh(ax=ax, transform=transform, cmap=cmap, **plot_kw)
-
+        im = plot_data.plot.pcolormesh(**plotting)
     else:
-        plot_kw.setdefault("levels", levels)
-        im = plot_data.plot.contourf(ax=ax, transform=transform, cmap=cmap, **plot_kw)
-
-    # add features
-    if features:
-        add_cartopy_features(ax, features)
-
-    if show_time:
-        if show_time is True:
-            plot_coords(
-                ax, plot_data, param="time", loc="lower right", backgroundalpha=1
-            )
-        elif isinstance(show_time, (str, tuple, int)):
-            plot_coords(ax, plot_data, param="time", loc=show_time, backgroundalpha=1)
-        else:
-            raise TypeError(" show_lat_lon must be a bool, string, int, or tuple")
-
-    set_plot_attrs(use_attrs, data, ax)
-
-    if frame is False:
-        ax.spines["geo"].set_visible(False)
-        if im.colorbar is not None:
+        plotting.setdefault("levels", levels)
+        im = plot_data.plot.contourf(**plotting)
+
+    if ax:
+        ax = add_features_map(
+            data,
+            ax,
+            use_attrs,
+            projection,
+            features,
+            geometries_kw,
+            show_time,
```

Comment:
I am really not sure what to do with show_time in facetgrids since it will probably become the title or be the same for all plots. Should we add it as a textbox somewhere outside plotting area (for exemple under plots)?

Reply:
maybe we just don't need it. We just not put it and put a warning.